src/KernelStack/SessionMiddleware.php

```diff
@@ -0,0 +1,108 @@
+<?php
+
+namespace Pyrite\KernelStack;
+
+use Symfony\Component\HttpFoundation\Cookie;
+use Symfony\Component\HttpFoundation\ParameterBag;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpFoundation\Session\Session;
+use Symfony\Component\HttpKernel\HttpKernelInterface;
+use Symfony\Component\HttpKernel\TerminableInterface;
+
+class SessionMiddleware implements HttpKernelInterface, TerminableInterface
+{
+    /**
+     * @var HttpKernelInterface
+     */
+    protected $app;
+
+    /**
+     * @var ParameterBag
+     */
+    protected $config;
+
+    /**
+     * SessionMiddleware constructor.
+     *
+     * @param HttpKernelInterface $app
+     * @param ParameterBag        $config
+     */
+    public function __construct(HttpKernelInterface $app, ParameterBag $config)
+    {
+        $this->app = $app;
+        $this->config = $config;
+    }
+
+    /**
+     * @param Request $request
+     * @param int     $type
+     * @param bool    $catch
+     *
+     * @return Response
+     * @throws \Exception
+     */
+    public function handle(Request $request, $type = self::MASTER_REQUEST, $catch = true)
+    {
+        $parameters = $this->config->get('cookie_parameters');
+
+        if(!isset($parameters[$this->config->get('current_locale')])){
```

Comment:
beware that the current_locale is evaneos centric, so is "cookie_parameters".
We should inject those as parameters for the session construction (that's why we have Factories in front of our stacks atm)